src/components/cards/VideoCardHorizontal.jsx

```diff
@@ -9,19 +9,19 @@ import { usePlaylist } from "../../context/PlaylistContext/PlaylistContext";
 
 
 const VideoCardHorizontal = ({details,location,playlistId,setOpenModal, setNewVideo}) => {
-  const {videoDispatch} = useVideo()
+  const searchVideo = (arr, video) => arr.includes(video);
+  const { videoState, videoDispatch } = useVideo();
   const {playlistDispatch} = usePlaylist()
   const {channelId, title, urls, creator, statistics,description } = details
 
   const handleAddToWatchLater = (event) => {
     event.stopPropagation();
-    if(location === "WatchLater"){
-      videoDispatch({type:"REMOVE_FROM_WATCHLATER",payload:details})
-      toast.success("Removed from Watch Later")
-    }
-    else{
-      videoDispatch({type:"ADD_TO_WATCHLATER",payload:details})
-      toast.success("Added to Watch Later")
+    if (searchVideo(videoState["watchLaterVideos"], details)) {
+      videoDispatch({ type: "REMOVE_FROM_WATCHLATER", payload: details });
```

Comment:
try to send payload as an `object` of key-value pair.

Reply:
noted will do